diff --git a/cdn.module b/cdn.module
index a004e81..2464413 100644
--- a/cdn.module
+++ b/cdn.module
@@ -466,11 +466,11 @@ function cdn_flush_caches() {
   cdn_get_blacklist(TRUE);
 
   // Clear aggregated CSS files for pages served via HTTP.
-  variable_del('drupal_css_cache_files_http');
+  variable_del('cdn_css_cache_files_http');
   file_scan_directory('public://cdn/css/http', '/.*/', array('callback' => 'drupal_delete_file_if_stale'));
 
   // Clear aggregated CSS files for pages served via HTTPS.
-  variable_del('drupal_css_cache_files_https');
+  variable_del('cdn_css_cache_files_https');
   file_scan_directory('public://cdn/css/https', '/.*/', array('callback' => 'drupal_delete_file_if_stale'));
 }
