diff --git a/page_manager/plugins/tasks/page.admin.inc b/page_manager/plugins/tasks/page.admin.inc
index 4e6e749..0161882 100644
--- a/page_manager/plugins/tasks/page.admin.inc
+++ b/page_manager/plugins/tasks/page.admin.inc
@@ -1056,6 +1056,7 @@ function page_manager_page_subtask_argument_ajax($step = NULL, $task_name = NULL
 
     $clone_state = $form_state;
     $clone_state['allow temp'] = TRUE;
+    $form = array();
     page_manager_page_form_argument($form, $clone_state);
     $form = drupal_build_form('page_manager_page_form_argument', $form_state);
 
