diff --git a/plugins/content_types/custom/custom.inc b/plugins/content_types/custom/custom.inc
index ba83eca..4853feb 100644
--- a/plugins/content_types/custom/custom.inc
+++ b/plugins/content_types/custom/custom.inc
@@ -423,7 +423,7 @@ function ctools_custom_content_type_edit_form_submit($form, &$form_state) {
     // Otherwise, just save values into $conf normally.
 
     foreach (array_keys($form_state['plugin']['defaults']) as $key) {
-      $form_state['conf'][$key] = $form_state['values'][$key];
+      $form_state['conf'][$key] = isset($form_state['values'][$key]) ? $form_state['values'][$key] : $form_state['plugin']['defaults'][$key];
     }
   }
 }
