diff --git a/modules/user/user.install b/modules/user/user.install
index 75fca65..0dfc058 100644
--- a/modules/user/user.install
+++ b/modules/user/user.install
@@ -405,7 +405,7 @@ function _update_7000_user_role_grant_permissions($rid, array $permissions, $mod
 function user_update_7000(&$sandbox) {
   $sandbox['#finished'] = 0;
   // Lower than DRUPAL_HASH_COUNT to make the update run at a reasonable speed.
-  $hash_count_log2 = 11;
+  $hash_count_log2 = 5;
   // Multi-part update.
   if (!isset($sandbox['user_from'])) {
     db_change_field('users', 'pass', 'pass', array('type' => 'varchar', 'length' => 128, 'not null' => TRUE, 'default' => ''));
