diff --git modules/field/modules/field_sql_storage/field_sql_storage.install modules/field/modules/field_sql_storage/field_sql_storage.install
index 783c293..f29fb56 100644
--- modules/field/modules/field_sql_storage/field_sql_storage.install
+++ modules/field/modules/field_sql_storage/field_sql_storage.install
@@ -34,6 +34,13 @@ function field_sql_storage_schema() {
 
   // Dynamic (data) tables.
   if (db_table_exists('field_config')) {
+
+    // field_sql_storage_schema() has been reported to run without field.module
+    // being loaded during D6 upgrades ran through drush. As a hotfix, the CRUD
+    // API file is explicitly included here. See http://drupal.org/node/893880.
+    // @todo Investigate and possibly fix drush update-db before a D7 release.
+    include_once(DRUPAL_ROOT . '/modules/field/field.crud.inc');
+
     $fields = field_read_fields(array(), array('include_deleted' => TRUE, 'include_inactive' => TRUE));
     drupal_load('module', 'field_sql_storage');
     foreach ($fields as $field) {
