diff --git a/includes/update.inc b/includes/update.inc
index 1eb7a1d..82e18af 100644
--- a/includes/update.inc
+++ b/includes/update.inc
@@ -388,7 +388,7 @@ function update_fix_d7_block_deltas(&$sandbox, $renamed_deltas, $moved_deltas) {
     $sandbox['max'] = db_query("SELECT COUNT(*) FROM {users} WHERE data IS NOT NULL")->fetchField();
   }
   // Now do the batch update of the user-specific block visibility settings.
-  $limit = 100;
+  /* $limit = 100;
   $result = db_select('users', 'u')
     ->fields('u', array('uid', 'data'))
     ->condition('uid', $sandbox['last_user_processed'], '>')
@@ -432,6 +432,8 @@ function update_fix_d7_block_deltas(&$sandbox, $renamed_deltas, $moved_deltas) {
     $sandbox['progress']++;
     $sandbox['last_user_processed'] = $row->uid;
   }
+  */
+$sandbox['progress'] = $sandbox['max'];
   // Indicate our current progress to the batch update system.
   if ($sandbox['progress'] < $sandbox['max']) {
     $sandbox['#finished'] = $sandbox['progress'] / $sandbox['max'];
