diff --git a/modules/user/user.install b/modules/user/user.install
index 8acd6ab..f5d20ea 100644
--- a/modules/user/user.install
+++ b/modules/user/user.install
@@ -234,6 +234,7 @@ function user_schema() {
       'access' => array('access'),
       'created' => array('created'),
       'mail' => array('mail'),
+      'picture' => array('picture'),
     ),
     'unique keys' => array(
       'name' => array('name'),
@@ -888,5 +889,12 @@ function user_update_7017() {
 }
 
 /**
+ * Add an index to {user}.picture.
+ */
+function user_update_7018() {
+  db_add_index('users', 'picture', array('picture'));
+}
+
+/**
  * @} End of "addtogroup updates-6.x-to-7.x"
  */
