diff --git a/modules/system/system.install b/modules/system/system.install
index c35c4f0..8313b8f 100644
--- a/modules/system/system.install
+++ b/modules/system/system.install
@@ -2176,6 +2176,7 @@ function system_update_7034() {
         'length' => 255,
         'not null' => TRUE,
         'default' => '',
+        'binary' => TRUE,
       ),
       'filemime' => array(
         'description' => "The file's MIME type.",
