diff --git a/feedback.admin.inc b/feedback.admin.inc
index 6640ef0..d5a7c21 100644
--- a/feedback.admin.inc
+++ b/feedback.admin.inc
@@ -222,6 +222,9 @@ function template_preprocess_feedback_entry(&$variables) {
   }
 
   $entry = $variables['elements']['#entry'];
+  if ($entry->uid) {
+    $entry->name = db_query("SELECT name FROM {users} WHERE uid = :uid", array(':uid' => $entry->uid))->fetchField();
+  }
 
   // Preprocess fields.
   field_attach_preprocess('feedback', $entry, $variables['elements'], $variables);
