diff --git a/geofield.install b/geofield.install
index 941625e..0d331e5 100644
--- a/geofield.install
+++ b/geofield.install
@@ -214,14 +214,16 @@ function geofield_update_7201(&$sandbox) {
 
       foreach ($result as $record) {
         $geom = geoPHP::load($record->wkt, 'wkt');
-        db_update($sandbox['geofield_tables'][$i]['table_name'])
-          ->fields(array(
-            $sandbox['geofield_tables'][$i]['field_name'] . '_geom' => $geom->out('wkb'),
-          ))
-          ->condition('entity_id', $record->entity_id)
-          ->condition('revision_id', $record->revision_id)
-          ->condition('delta', $record->delta)
-          ->execute();
+        if ($geom) {
+          db_update($sandbox['geofield_tables'][$i]['table_name'])
+            ->fields(array(
+              $sandbox['geofield_tables'][$i]['field_name'] . '_geom' => $geom->out('wkb'),
+            ))
+            ->condition('entity_id', $record->entity_id)
+            ->condition('revision_id', $record->revision_id)
+            ->condition('delta', $record->delta)
+            ->execute();
+        }
 
         $sandbox['batch_count']++;
         $sandbox['count']++;
