diff --git a/og/plugins/og_migrate/group_membership.inc b/og/plugins/og_migrate/grou
index 55b3175..e8c683a 100644
--- a/og/plugins/og_migrate/group_membership.inc
+++ b/og/plugins/og_migrate/group_membership.inc
@@ -46,6 +46,10 @@ function og_og_migrate_group_membership(&$context) {
       field_create_instance($instance);
     }
 
+    // It is probably already added, but make sure there is og-audience field
+    // attached to the user.
+    og_create_field(OG_AUDIENCE_FIELD, 'user', 'user');
+
     // Calculate max items.
     $query = new EntityFieldQuery();
     $context['sandbox']['max'] = $query
     diff --git a/og/og_migrate/plugins/og_migrate/upgrade_group.inc b/og/og_migrate/plugins/og_migrate/upgrade_group.inc
     index b685bd6..652a2a7 100644
     --- a/og/og_migrate/plugins/og_migrate/upgrade_group.inc
     +++ b/og/og_migrate/plugins/og_migrate/upgrade_group.inc
     @@ -95,6 +95,13 @@ function og_migrate_og_migrate_upgrade_group(&$context) {
        }

        if ($context['sandbox']['progress'] != $context['sandbox']['max']) {
     -    $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
     +    // Prevent batch from getting stuck and never finishing.
     +    if (isset($context['sandbox']['last_progress']) && $context['sandbox']['last_progress'] == $context['sandbox']['progress']) {
     +      $context['finished'] = 1;
     +    }
     +    else {
     +      $context['sandbox']['last_progress'] = $context['sandbox']['progress'];
     +      $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
     +    } 
        }
      }
     \ No newline at end of file
     diff --git a/og/og_migrate/plugins/og_migrate/upgrade_group_content.inc b/og/og_migrate/plugins/og_migrate/upgrade_group_content.inc
     index 8b7a136..0ada7e9 100644
     --- a/og/og_migrate/plugins/og_migrate/upgrade_group_content.inc
     +++ b/og/og_migrate/plugins/og_migrate/upgrade_group_content.inc
     @@ -72,6 +72,13 @@ function og_migrate_og_migrate_upgrade_group_content(&$context) {
          $context['sandbox']['progress']++;
        }
        if ($context['sandbox']['progress'] != $context['sandbox']['max']) {
     -    $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
     +    // Prevent batch from getting stuck and never finishing.
     +    if (isset($context['sandbox']['last_progress']) && $context['sandbox']['last_progress'] == $context['sandbox']['progress']) {
     +      $context['finished'] = 1;
     +    }
     +    else {
     +      $context['sandbox']['last_progress'] = $context['sandbox']['progress'];
     +      $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
     +    }
        }
      }
     diff --git a/og/og_migrate/plugins/og_migrate/upgrade_group_description.inc b/og/og_migrate/plugins/og_migrate/upgrade_group_description.inc
     index 1547c32..9ddc911 100644
     --- a/og/og_migrate/plugins/og_migrate/upgrade_group_description.inc
     +++ b/og/og_migrate/plugins/og_migrate/upgrade_group_description.inc
     @@ -84,8 +84,15 @@ function og_migrate_og_migrate_upgrade_group_description(&$context) {
            $context['sandbox']['progress']++;
          }
        }

        if ($context['sandbox']['progress'] != $context['sandbox']['max']) {
     -    $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
     +    // Prevent batch from getting stuck and never finishing.
     +    if (isset($context['sandbox']['last_progress']) && $context['sandbox']['last_progress'] == $context['sandbox']['progress']) {
     +      $context['finished'] = 1;
     +    }
     +    else {
     +      $context['sandbox']['last_progress'] = $context['sandbox']['progress'];
     +      $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
     +    }
        }
      }
     diff --git a/og/og_migrate/plugins/og_migrate/upgrade_user.inc b/og/og_migrate/plugins/og_migrate/upgrade_user.inc
     index 53bf5fd..a9151d6 100644
     --- a/og/og_migrate/plugins/og_migrate/upgrade_user.inc
     +++ b/og/og_migrate/plugins/og_migrate/upgrade_user.inc
     @@ -63,19 +63,26 @@ function og_migrate_og_migrate_upgrade_user(&$context) {

          $account = user_load($record->uid);
          // Make sure entity isn't already associated with group.
     -    if (!og_is_member($group->gid, 'user', $account)) {
     +    if ($group->gid && $account->uid && !og_is_member($group->gid, 'user', $account)) {
            $values = array('entity' => $account, 'state' => $state);
            og_group($group->gid, $values);
     +      if ($record->is_admin) {
     +        og_role_grant($group->gid, $record->uid, $context['sandbox']['rid']);
     +      }
          }

     -    if ($record->is_admin) {
     -      og_role_grant($group->gid, $record->uid, $context['sandbox']['rid']);
     -    }
          $context['sandbox']['last'] = $record->upgrade_id;
          $context['sandbox']['progress']++;
        }

        if ($context['sandbox']['progress'] != $context['sandbox']['max']) {
     -    $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
     +    // Prevent batch from getting stuck and never finishing.
     +    if (isset($context['sandbox']['last_progress']) && $context['sandbox']['last_progress'] == $context['sandbox']['progress']) {
     +      $context['finished'] = 1;
     +    }
     +    else {
     +      $context['sandbox']['last_progress'] = $context['sandbox']['progress'];
     +      $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
     +    }
        }
      }
     \ No newline at end of file
     diff --git a/og/og_ui/plugins/og_migrate/upgrade_group_visibility.inc b/og/og_ui/plugins/og_migrate/upgrade_group_visibility.inc
     index edb77a3..1745d32 100644
     --- a/og/og_ui/plugins/og_migrate/upgrade_group_visibility.inc
     +++ b/og/og_ui/plugins/og_migrate/upgrade_group_visibility.inc
     @@ -132,16 +132,25 @@ function og_ui_og_migrate_upgrade_group_visibility(&$context) {
              // Since the node is already a group we have to manually override
              // the roles, as og_group() isn't called.
              $roles = og_roles_override($group->gid);
     -        $roles = array_flip($roles);
     +        if (count($roles)) {
     +          $roles = array_flip($roles);

     -        // Set the new permissions.
     -        og_role_change_permissions($roles[OG_ANONYMOUS_ROLE], $anon_permissions);
     -        og_role_change_permissions($roles[OG_AUTHENTICATED_ROLE], $auth_permissions);
     +          // Set the new permissions.
     +          og_role_change_permissions($roles[OG_ANONYMOUS_ROLE], $anon_permissions);
     +          og_role_change_permissions($roles[OG_AUTHENTICATED_ROLE], $auth_permissions);
     +        }
            }
          }
        }

        if ($context['sandbox']['progress'] != $context['sandbox']['max']) {
     -    $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
     +    // Prevent batch from getting stuck and never finishing.
     +    if (isset($context['sandbox']['last_progress']) && $context['sandbox']['last_progress'] == $context['sandbox']['progress']) {
     +      $context['finished'] = 1;
     +    }
     +    else {
     +      $context['sandbox']['last_progress'] = $context['sandbox']['progress'];
     +      $context['finished'] = $context['sandbox']['progress'] / $context['sandbox']['max'];
     +    }
        }
      }
     \ No newline at end of file