diff --git a/og_ui/og_ui.admin.inc b/og_ui/og_ui.admin.inc
index 4d99da6..fdf862e 100644
--- a/og_ui/og_ui.admin.inc
+++ b/og_ui/og_ui.admin.inc
@@ -44,7 +44,10 @@ function og_ui_group_admin_overview($entity_type, $etid) {
     $item['localized_options'] = array();
   }
 
-  $form['admin_block_content'] = array('#markup' => theme('admin_block_content', array('content' => $items)));
+  $form['admin_block_content'] = array(
+    '#theme' => 'admin_block_content', 
+    '#content' => $items,
+  );
 
   return $form;
 }
