diff --git a/panels.module b/panels.module
index 3ea7e67..a6fec5d 100644
--- a/panels.module
+++ b/panels.module
@@ -1193,9 +1193,14 @@ function template_preprocess_panels_pane(&$vars) {
   $vars['id'] = '';
 
   // Add some usable classes based on type/subtype
-  ctools_include('cleanstring');
-  $type_class = $content->type ? 'pane-'. ctools_cleanstring($content->type, array('lower case' => TRUE)) : '';
-  $subtype_class = $content->subtype ? 'pane-'. ctools_cleanstring($content->subtype, array('lower case' => TRUE)) : '';
+  if (isset($content->type)) {
+    ctools_include('cleanstring');
+    $type_class = $content->type ? 'pane-'. ctools_cleanstring($content->type, array('lower case' => TRUE)) : '';
+    $subtype_class = $content->subtype ? 'pane-'. ctools_cleanstring($content->subtype, array('lower case' => TRUE)) : '';
+  }
+  else {
+    $subtype_class = $type_class = '';
+  }
 
   // Sometimes type and subtype are the same. Avoid redundant classes.
   $vars['classes_array'][] = $type_class;
