diff --git a/includes/add-content.inc b/includes/add-content.inc
index 128c78c..7e6d8ac 100644
--- a/includes/add-content.inc
+++ b/includes/add-content.inc
@@ -53,12 +53,10 @@ function template_process_panels_add_content_modal(&$vars) {
     $titles = array_keys($content);
     natcasesort($titles);
 
-    // This will default to 2 columns in the theme definition but could
-    // theoretically be changed by a preprocess.
-    $columns = $vars['column_count'];
-    foreach (range($columns) as $column) {
-      $vars['columns'][$column] = '';
-    }
+    // This will default to 2 columns in the theme definition but could be
+    // changed by a preprocess. Ensure there is at least one column.
+    $columns = ($vars['column_count'] > 0) ? $vars['column_count'] : 1;
+    $vars['columns'] = array_fill(1, $columns, '');
 
     $col_size = count($titles) / $columns;
     $count = 0;
@@ -76,7 +74,7 @@ function template_process_panels_add_content_modal(&$vars) {
  */
 function template_preprocess_panels_add_content_link(&$vars) {
   $vars['title'] = filter_xss_admin($vars['content_type']['title']);
-  $vars['description'] = isset($vars['content_type']['description']) ? $vars['content_type']['description'] : $title;
+  $vars['description'] = isset($vars['content_type']['description']) ? $vars['content_type']['description'] : $vars['title'];
   $vars['icon'] = ctools_content_admin_icon($vars['content_type']);
   $vars['url'] = $vars['renderer']->get_url('add-pane', $vars['region'], $vars['content_type']['type_name'], $vars['content_type']['subtype_name']);
 
diff --git a/templates/panels-add-content-modal.tpl.php b/templates/panels-add-content-modal.tpl.php
index 16d9e5e..5041ff0 100644
--- a/templates/panels-add-content-modal.tpl.php
+++ b/templates/panels-add-content-modal.tpl.php
@@ -24,6 +24,7 @@
     </div>
   <?php endif; ?>
 
+  <?php if (!empty($columns)): ?>
   <div class="panels-section-columns">
     <?php foreach ($columns as $column_id => $column): ?>
       <div class="panels-section-column panels-section-column-<?php print $column_id; ?> ">
@@ -33,5 +34,5 @@
       </div>
     <?php endforeach; ?>
   </div>
-  <?php print $category; ?>
+  <?php endif; ?>
 </div>
