diff --git a/sites/all/modules/redirect/redirect.module b/sites/all/modules/redirect/redirect.module
index d7f5349..3447367 100644
--- a/sites/all/modules/redirect/redirect.module
+++ b/sites/all/modules/redirect/redirect.module
@@ -201,27 +201,6 @@ function redirect_menu() {
 }
 
 /**
- * Implements hook_url_inbound_alter().
- */
-function redirect_url_inbound_alter(&$path, $original_path, $path_language) {
-  // Redirect to canonical URLs.
-  if ($path && variable_get('redirect_canonical', 1)) {
-    $alias = drupal_get_path_alias($path, $path_language);
-    if ($alias != $path && $alias != $original_path) {
-      //return redirect_redirect(array('redirect' => $alias, 'type' => 'global'));
-    }
-
-    // Redirect from default entity paths to the proper entity path.
-    if ($path_entity = redirect_load_entity_from_path($path)) {
-      $uri = entity_uri($path_entity['entity_type'], $path_entity['entity']);
-      if ($path != $uri['path']) {
-        //return redirect_redirect(array('redirect' => $uri['path'], 'redirect_options' => $uri['options'], 'type' => 'global'));
-      }
-    }
-  }
-}
-
-/**
  * Implements hook_entity_info_alter().
  */
 function redirect_entity_info_alter(&$info) {
