diff --git a/reroute_email.module b/reroute_email.module
index d4510fc..97e3d4d 100644
--- a/reroute_email.module
+++ b/reroute_email.module
@@ -47,7 +47,7 @@ function reroute_email_settings() {
     '#default_value' => variable_get(REROUTE_EMAIL_ADDRESS, ini_get('sendmail_from')),
     '#cols'          => 35,
     '#rows'          => 5,
-    '#description'   => t('Provide a list of email addresses to pass through. Any email addresses not on this list will be rerouted to the first address on the list.')
+    '#description'   => t('Provide a list of email addresses to pass through. Any email addresses not on this list will be rerouted to the first address on the list. The use of wildcards such as *@example.com is acceptable.'),
   );
 
   $form[REROUTE_EMAIL_ENABLE] = array(
@@ -98,7 +98,22 @@ function reroute_email_mail_alter(&$message) {
     // Split the address string on whitespace, ignoring any empty results
     $addresslist = preg_split('/\s/', variable_get(REROUTE_EMAIL_ADDRESS, ini_get('sendmail_from')), -1, PREG_SPLIT_NO_EMPTY);
 
-    if (in_array($to, $addresslist)) {
+    foreach ($addresslist as $address) {
+      // Allow use of * wildcards.
+      if (substr($address, 0, 2) == '*@') {
+        $to_parts = explode('@', $to);
+        $address_parts = explode('@', $address);
+        if ($to_parts[1] == $address_parts[1]) {
+          $passthrough = TRUE;
+        }
+      }
+      else {
+        if ($to == $address) {
+          $passthrough = TRUE;
+        }
+      }
+    }
+    if (isset($passthrough)) {
       // To address is in the pass-through list, let it pass through
       $message['to'] = $to;
     }
