diff --git a/plugins/views_plugin_display.inc b/plugins/views_plugin_display.inc
index 32c560b..96562e2 100644
--- a/plugins/views_plugin_display.inc
+++ b/plugins/views_plugin_display.inc
@@ -2214,12 +2214,12 @@ class views_plugin_display extends views_plugin {
         }
         break;
 
+      case 'link_display':
+        $this->set_option('link_url', $form_state['values']['link_url']);
       case 'title':
       case 'css_class':
-      case 'link_display':
       case 'display_comment':
         $this->set_option($section, $form_state['values'][$section]);
-        $this->set_option('link_url', $form_state['values']['link_url']);
         break;
       case 'field_language':
         $this->set_option('field_language', $form_state['values']['field_language']);
