diff --git a/webform.module b/webform.module
index 2fffd0c..23d44d9 100644
--- a/webform.module
+++ b/webform.module
@@ -3365,7 +3356,7 @@ function webform_format_email_subject($subject, $node = NULL, $submission = NULL
  */
 function _webform_components_tree_build($src, &$tree, $parent, &$page_count) {
   foreach ($src as $cid => $component) {
-    if ($component['pid'] == $parent) {
+    if ((int) $component['pid'] === $parent) {
       _webform_components_tree_build($src, $component, $cid, $page_count);
       if ($component['type'] == 'pagebreak') {
         $page_count++;
