diff --git a/views/webform.views.inc b/views/webform.views.inc
index f8e7c6f..67718ed 100644
--- a/views/webform.views.inc
+++ b/views/webform.views.inc
@@ -28,6 +28,10 @@ function webform_views_data() {
       'title' => t('Sid'),
       'handler' => 'views_handler_filter_numeric',
     ),
+    'argument' => array(
+      'title' => t('Sid'),
+      'handler' => 'views_handler_argument_numeric',
+    ),
     'sort' => array(
       'handler' => 'views_handler_sort',
     ),
